pimmi/cli/__main__.py

```diff
@@ -213,40 +215,62 @@ def query( image_dir, index_dir,output, nb_per_split,output_template, simple, nb
     logger.info("total number of queries " + str(len(images)))
     # images = images.sort_values(by=constants.dff_image_path)
     images.sort(key=lambda x: x[constants.dff_image_path])
-    if nb_per_split:
+    queries = tbx.split_pack(images, nb_per_split)
+
+    list_files = []
+    if nb_per_files:
         if not output_template:
-            logger.error("Please restart pimmi query --nb-per-split with '--filename-template' or do not use --nb-per-split.")
-            sys.exit(1)
-        queries = tbx.split_pack(images, nb_per_split)
-        for i, pack in enumerate(queries):
-            pack_result_file = output_template+str(i)+".csv" if output else sys.stdout
-            #logger.info("query " + str(len(pack[constants.dff_pack_files])) + " files from pack " +
-                        #str(pack[constants.dff_pack_id]) + " -> " + pack_result_file)
-            query_result = query_index_mt(
-                index,
-                pack[constants.dff_pack_files],
-                image_dir,
-                pack=pack[constants.dff_pack_id]
-            )
-            if query_result:
-                query_result.to_csv(pack_result_file)
-    else :
-        pack_result_file = output if output else sys.stdout
-        #logger.info("query " + str(len(pack[constants.dff_pack_files])) + " files from pack " +
-                    #str(pack[constants.dff_pack_id]) + " -> " + pack_result_file)
-        query_result = query_index_mt(
-            index,
-            images,
-            image_dir,
-            pack=0
-        )
-        if query_result:
-            query_result.to_csv(pack_result_file)
+                logger.error("Please restart pimmi query --nb-per-files with '--output-template' or do not use --nb-per-files.")
+                sys.exit(1)
+        else : 
+            for i in range(math.ceil(len(images)/nb_per_files)):
+                list_files.append(output_template+str(i)+".csv")
+    else:
+        if output :
+            list_files.append(output)
+        else :
+            list_files.append(sys.stdout)  
+        nb_per_files = len(images)
```

Comment:
The total number of lines can be max nb_images**2 but it can also be less if no match is found for an image --> you don't know in advance the number of lines